src/EditorFeatures/Core/Implementation/Classification/SemanticClassificationUtilities.cs

```diff
@@ -173,16 +173,26 @@ internal static class SemanticClassificationUtilities
             List<ClassifiedSpan> classifiedSpans,
             CancellationToken cancellationToken)
         {
-            var workspace = document.Project.Solution.Workspace;
+            // Note: we do this work in a Task.Run to ensure that nothing we do (sync or async) ends up causing blocking
+            // on the UI thread inside of IWorkspaceStatusService.  This is necessary as synchronous tagging will cause
+            // us to make an explicit .Wait call on the tagging tasks.  If this thread in any way ended up blocking on 
+            // the UI thread, we would deadlock.
             var fullyLoadedStateTask = s_workspaceToFullyLoadedStateTask.GetValue(
-                workspace, w =>
+                document.Project.Solution.Workspace,
+                w => Task.Run(async () =>
                 {
-                    var workspaceLoadedService = workspace.Services.GetRequiredService<IWorkspaceStatusService>();
-                    return workspaceLoadedService.WaitUntilFullyLoadedAsync(CancellationToken.None);
-                });
+                    var workspaceLoadedService = w.Services.GetRequiredService<IWorkspaceStatusService>();
+                    await workspaceLoadedService.WaitUntilFullyLoadedAsync(CancellationToken.None).ConfigureAwait(false);
```

Comment:
Did this need to be async? Task.Run() should have worked fine if the lambda is just directly returning the Task.